content/docs/hoon/reference/stdlib/5d.md

```diff
@@ -4,4 +4,318 @@ weight = 46
+```
+
+#### Examples
+
+```
+> (rash '(add 1 1)' vest)
+[%cncl p=[%wing p=~[%add]] q=~[[%sand p=%ud q=1] [%sand p=%ud q=1]]]
+
+> (rash '%+  add\0a  1\0a1' vest)
+[%cnls p=[%wing p=~[%add]] q=[%sand p=%ud q=1] r=[%sand p=%ud q=1]]
+```
+
+---
+
+### `++vice`
+
+Parse wide-form hoon
+
+Parse `txt`, a `knot` containing wide-form hoon, to `hoon`.
+
+#### Accepts
+
+`txt` is a `@ta`.
+
+#### Produces
+
+`hoon`.
+
+#### Source
+
+```hoon
+++  vice
+  |=  txt=@ta
+  ^-  hoon
+  (rash txt wide:vast)
+```
+
+#### Examples
+
+```
+> (vice '(add 1 1)')
+[%cncl p=[%wing p=~[%add]] q=~[[%sand p=%ud q=1] [%sand p=%ud q=1]]]
+
+> (vice '%+  add\0a  1\0a1')
+{1 3}
+syntax error
+```
+
+---
+
+## `++make`
+
+Compile cord to nock
+
+Compile `txt`, a cord containing hoon, to `nock`.
+
+#### Accepts
+
+`txt` is a `@`.
+
+#### Produces
+
+`nock`.
+
+#### Source
+
+```hoon
+++  make
+  |=  txt=@
+  q:(~(mint ut %noun) %noun (ream txt))
+```
+
+#### Examples
+
+```
+> (make '[. . .]')
+[p=[%0 p=1] q=[p=[%0 p=1] q=[%0 p=1]]]
+
+> .*(42 (make '[. . .]'))
+[42 42 42]
+```
+
+---
+
+## `++rain`
+
+Parse with % path
+
+Parse `txt`, a cord containing hoon, to `hoon`. Any `%` path short-hands are
+replaced with the path given in `bon`. If parsing fails, `bon` is also printed
+to the terminal.
+
+#### Accepts
+
+`bon` is a `path`.
+
+`txt` is an atom.
+
+#### Produces
+
+`hoon`.
+
+#### Source
+
+```hoon
+++  rain
+  |=  [bon=path txt=@]
+  ^-  hoon
+  =+  vaz=vast
+  ~|  bon
+  (scan (trip txt) (full (ifix [gay gay] tall:vaz(wer bon))))
+```
+
+#### Examples
+
+```
+> (rain /a/b/c '%')
+[%clsg p=~[[%sand p=%ta q=97] [%sand p=%ta q=98] [%sand p=%ta q=99]]]
+
+> !<(path (slap !>(~) (rain /a/b/c '%')))
+/a/b/c
+
+> (rain / '(add 1 1)')
+[%cncl p=[%wing p=~[%add]] q=~[[%sand p=%ud q=1] [%sand p=%ud q=1]]]
+
+> !<(@ud (slap !>(add=add) (rain / '(add 1 1)')))
+2
+```
+
+---
+
+## `++ream`
+
+Parse cord to hoon
+
+Parse cord `txt` to `hoon`.
+
+#### Accepts
+
+`txt` is an atom.
+
+#### Produces
+
+`hoon`.
+
+#### Source
+
+```hoon
+++  ream
+  |=  txt=@
+  ^-  hoon
+  (rash txt vest)
+```
+
+#### Examples
+
+```
+> !<(tape (slap !>(~) (ream '"foobar"')))
+"foobar"
+```
+
+---
+
+## `++reck`
+
+Parse hoon file
+
+Parse `bon` to `hoon`. `bon` is a `path` to a .hoon file without the mark.
+
+#### Accepts
+
+`bon` is a `path`.
+
+#### Produces
+
+`hoon`.
+
+#### Source
+
+```hoon
+++  reck
+  |=  bon=path
+  (rain bon .^(@t %cx (weld bon `path`[%hoon ~])))
+```
+
+#### Examples
+
+```
+> (reck %/gen/code)
+[ %clhp
+  p=[%rock p=%tas q=7.954.803]
+    q
+  [ %brts
+      p
+    [ %bccl
+        p
+      [   i
+        [ %bccl
+            p
+          [   i
+            [%bcts p=term=%now q=[%base p=[%atom p=~.da]]]
+              t
+......(truncated for brevity)..........
+```
+
+---
+
+## `++ride`
+
+End-to-end compiler
+
+Parse and compile cord `txt`, producing a pair of its `type` and compiled
+`nock`. The `typ` argument specifies the `type` of the subject.
+
```

Comment:
`+ride` is somewhat special and I think that should be pointed out. It's the entrypoint to the compiler, as the first noncommented line is `=<  ride`.  Maybe this is implicit by saying "end-to-end" but only if you already know that. Just one sentence ought to be enough.

Also the `txt=@` thing

Reply:
can you explain the implications of this? It's not clear to me what this practically means